src/frontend/src/components/toast.ts

```diff
@@ -1,60 +1,67 @@
-import { render } from "lit-html";
+import { html, render, TemplateResult } from "lit-html";
+import { asyncReplace } from "lit-html/directives/async-replace.js";
+import { repeat } from "lit/directives/repeat.js";
 import { TemplateElement } from "../utils/lit-html";
-import { irregularity } from "./irregularity";
+import { Chan } from "../utils/utils";
+import { closeIcon, warningIcon } from "./icons";
 
-export const removeToast = (toast: HTMLElement): void => {
-  toast.addEventListener("animationend", () => {
-    toast.remove();
-  });
-  toast.classList.add("c-toast--closing");
-};
-
-const createToast = (message: TemplateElement): HTMLDivElement => {
-  const toastEl = document.createElement("div");
-  toastEl.classList.add("c-toast");
-  render(
-    irregularity({
-      message,
-      closeFn: () => removeToast(toastEl),
-    }),
-    toastEl
-  );
-  return toastEl;
-};
-
-/**
- * Get the container for the toasts, or create it if it doesn't exist.
- * @returns the container reference for the toasts
- */
-const getOrCreateToaster = (): Element | null => {
-  const toasterEl = document.querySelector("[data-toaster]");
-  if (!toasterEl) {
-    // generic toaster that will be appended to the body
-    const toasterEl = document.createElement("div");
-    toasterEl.classList.add("c-toasts");
-    // container for the toasts. Two reasons for this:
-    // 1. we can reuse the l-container class for styling
-    // 2. the toasts will only be as wide as the container (user can still interact with the rest of the page)
-    const containerEl = document.createElement("div");
-    containerEl.classList.add("l-container");
-    // Attribute used to retrieve the element to append toasts to
-    containerEl.setAttribute("data-toasts", "");
-    toasterEl.setAttribute("data-toaster", "");
-    toasterEl.appendChild(containerEl);
-    document.body.appendChild(toasterEl);
-    return containerEl;
-  }
-  return toasterEl.querySelector("[data-toasts]");
-};
+// A toast element containing a message, and the (static) list of all toasts. The `Toast` element
+// must be an object because we (lit) use object-references in `repeat` to figure out which DOM elements
+// to replace.
+type Toast = { message: TemplateElement };
+const toasts: Toast[] = []; // the "toasts" singleton
```

Comment:
@peterpeterparker how bad is this in your eyes?

Reply:
Sounds good to me

Reply:
it's kind of a store, should we call it `toastStore`? Just an idea, no strong opinion

Reply:
That's a good idea, let me do that!